execution/gethexec/executionengine.go

```diff
@@ -526,6 +528,9 @@ func (s *ExecutionEngine) sequenceTransactionsWithBlockMutex(header *arbostypes.
 	if err != nil {
 		return nil, err
 	}
+	if s.addressFilter != nil {
+		statedb.SetAddressFilter(s.addressFilter)
```

Comment:
we also need to apply the filter in sequenceDelayedMessageWithBlockMutex (and test it)